crates/host/src/wasmbus/handler.rs

```diff
@@ -456,6 +460,642 @@ impl Messaging for Handler {
     }
 }
 
+struct MessagingClient {
+    name: Box<str>,
+}
+
+#[async_trait]
+impl MessagingClient0_3 for MessagingClient {
+    async fn disconnect(&mut self) -> anyhow::Result<Result<(), messaging0_3_0::types::Error>> {
+        Ok(Ok(()))
+    }
+
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+}
+
+pub enum Message {
+    Nats(async_nats::Message),
+}
```

Comment:
I was surprised to see this as a concrete implementation of _just_ NATS here, I expected to see the same pattern we use elsewhere where we send a wRPC message to some wasmcloud:messaging@0.3.0 provider. Is that not applicable for this new interface, or is this just a sample implementation?

Reply:
Both approaches are supported, this is a message originating from a builtin provider.

Reply:
Thanks for the clarity!